aries_vcx/src/handlers/out_of_band/mod.rs

```diff
@@ -43,8 +42,41 @@ pub struct OutOfBand {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub handshake_protocols: Option<Vec<MessageType>>,
     pub services: Vec<ServiceResolvable>,
-    #[serde(rename = "requests~attach", skip_serializing_if = "Attachments::is_empty")]
+    #[serde(rename = "requests~attach")]
     pub requests_attach: Attachments,
 }
 
 a2a_message!(OutOfBand);
+
+impl OutOfBand {
+    pub fn to_string(&self) -> VcxResult<String> {
+        serde_json::to_string(&self)
+            .map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot serialize out of band message: {:?}", err)))
+    }
+
+    pub fn from_string(oob_data: &str) -> VcxResult<OutOfBand> {
+        serde_json::from_str(oob_data)
+            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize out of band message: {:?}", err)))
+    }
+}
+
+// TODO: Add more tests
+#[cfg(test)]
```

Comment:
We could add test to compare serialized OOB against expected string / serde JSON.

Reply:
Added a verification to the existing test.